### DISCUSSION TITLE
how to update kernel,device tree partition ?

### DISCUSSION THREAD

**Discussion by gitgonewithwind** [rauc/rauc#1007] [Q&A]
hi all,
sorry to bother you again.
I can update rootfs, but I am not clear  how to update kernel,device tree partition ?

my sd card has 3 partitions and an invisible uboot partition 
mmcblk0p1     boot partition, kernel(Image) , uEnv.txt    dtb .. are here    
mmcblk0p2   rootfs.0                     
mmcblk0p3  rootfs.1
now I can update (switch) between rootfs.0 and rootfs.1 with below system.conf

``` ini
[system]                                                                       
compatible=maaxboard-mini-zeus                                                 
bootloader=uboot                                                               
                                                                           
[keyring]                                                                       
path=/etc/rauc/ca.cert.pem
                                                 
[slot.rootfs.0]                                                                 
device=/dev/mmcblk0p2                                                           
type=ext4                                                                       
bootname=system0                                                                
                                                                                
[slot.rootfs.1]                                                                
device=/dev/mmcblk0p3                                                           
type=ext4                                                                       
bootname=system1        
```

my lite-image-bundle-rauc.bb  file in yocto.

```
inherit bundle

RAUC_BUNDLE_COMPATIBLE ?= "maaxboard-mini-zeus"

RAUC_BUNDLE_SLOTS ?= "rootfs"

RAUC_SLOT_rootfs ?= "lite-image"

RAUC_KEY_FILE = "${COREBASE}/../meta-maaxboard/images/rauc-files/development-1.key.pem"
RAUC_CERT_FILE = "${COREBASE}/../meta-maaxboard/images/rauc-files/development-1.cert.pem"
```

but how to  update kernel,device tree partition ?mmcblk0p1  (Maybe  I can add a redundancy partition).
but how to write  in  my lite-image-bundle-rauc.bb  at line  RAUC_BUNDLE_SLOTS ?= "rootfs" ? use "boot" replace "rootfs"

or add  
some lines in system.conf,like this:

``` ini
[slot.rootfs.3]                                                                 
device=/dev/mmcblk0p1                                                           
type=fat32                                                                       
bootname=system3       

[slot.rootfs.4]                                                                 
device=/dev/mmcblk0p??(    redundancy partition I will add )                                                        
type=fat32                                                                       
bootname=system4 
```

right ?

Can I update kernel like this:
I put the mmcblk0p1 partition   new files: kernel Image,dtb .. into rootfs to generate the .raucb file,
after updating successfally (runc install .raucb),
mount the updated rootfs, 
and copy the kernel Image,dtb .. files   into  mmcblk0p1  to replace the older files)(so I do not need to add a redundancy partition for mmcblk0p1)

any help? thank you very much.

**Comment by ejoerns** [comment]
@gitgonewithwind please do not open any further Issues for usage questions. Use GitHub Discussions instead.

Also please take your time to format and formulate your questions in a way that people willing to help (and invest their time) have a chance to read them, thanks!

For updating your boot partition, you have many options, but I do not know which requirements are set in your system and which are not.
Also your examples provided suggest that you should have another look at how RAUC slots and slot grouping works:

https://rauc.readthedocs.io/en/latest/basic.html#slots

Also note that RAUC updates entire partitions/images. Thus if you have multiple individual files, you should either first create an image from them or install them into individual slots.

If you want to learn about how bootloader slots can be defined and updated atomically, refer to

https://rauc.readthedocs.io/en/latest/advanced.html#updating-the-bootloader

The other questions might be more appropriate to be asked in meta-rauc.
